fstwrapper/src/set.rs

```diff
@@ -146,13 +146,28 @@ pub extern "C" fn fst_set_make_opbuilder(ptr: *mut Set) -> *mut set::OpBuilder<'
 }
 make_free_fn!(fst_set_opbuilder_free, *mut set::OpBuilder);
 
+#[no_mangle]
+pub extern "C" fn fst_set_make_opstreambuilder(ptr: *mut set::Stream) -> *mut set::OpBuilder<'static> {
+    let stream = ref_from_ptr!(ptr);
```

Comment:
@davidblewett I think you want `stream` to be a `set::Stream` here, but it looks like it's a `&set::Stream`. In particular, this requires passing ownership of the stream into this function, so I imagine you want something like `let stream = unsafe { *Box::from_raw(ptr) };`. Similarly for the function below.

I haven't audited this code thoroughly, but I might be concerned about memory leaks here. e.g., What's responsible for freeing the memory of an opbuilder?

Reply:
https://github.com/jbaiter/python-rust-fst/blob/master/rust_fst/set.py#L60-L61 :
 
        # NOTE: No need for `ffi.gc`, since the struct will be free'd
        #       once we call union/intersection/difference

Reply:
Ah, right! Nice catch.

Reply:
@BurntSushi I tried updating to use `*mut &set::Stream`, but the result didn't change:

```
  Compiling fst-wrapper v0.3.0 (file:///home/dblewett/src/python-rust-fst/fstwrapper)                          
error[E0277]: the trait bound `for<'a> &fst::set::Stream<'_>: fst::Streamer<'a>` is not satisfied               
   --> src/set.rs:152:36                                
    |                                                   
152 |     let ob = set::OpBuilder::new().add(stream);   
    |                                    ^^^ the trait `for<'a> fst::Streamer<'a>` is not implemented for `&fst::set::Stream<'_>`                                                                                               
    |                                                   
    = help: the following implementations were found:   
              <fst::set::Stream<'s, A> as fst::Streamer<'a>>                                                    

error[E0277]: the trait bound `for<'a> &fst::set::Stream<'_>: fst::Streamer<'a>` is not satisfied               
   --> src/set.rs:168:8                                 
    |                                                   
168 |     ob.push(stream);                              
    |        ^^^^ the trait `for<'a> fst::Streamer<'a>` is not implemented for `&fst::set::Stream<'_>`          
    |                                                   
    = help: the following implementations were found:   
              <fst::set::Stream<'s, A> as fst::Streamer<'a>>                                                    

error: aborting due to 2 previous errors                

error: Could not compile `fst-wrapper`.   
```